resources/windows/wix/service-filter.xslt.erb

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="1.0"
+ xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+ xmlns:wix="http://schemas.microsoft.com/wix/2006/wi">
+  <!-- https://ahordijk.wordpress.com/2013/03/26/automatically-add-references-and-content-to-the-wix-installer/ -->
+  <!-- http://www.chriskonces.com/using-xslt-with-heat-exe-wix-to-create-windows-service-installs/ -->
+  <xsl:output method="xml" indent="yes" />
+  <!--<xsl:strip-space elements="*"/>-->
+  <xsl:template match="@*|node()">
+    <xsl:copy>
+      <xsl:apply-templates select="@*|node()"/>
+    </xsl:copy>
+  </xsl:template>
+  <!-- Filter out all Service File Executables from the Harvest (heat) run as these are specified in the transformed service.components.wxs file -->
+  <%- get_services.each do |service| -%>
+      <xsl:template match="wix:Component[wix:File[@Source='<%= service.options.first[:file][:source] %>']]" />

```

Comment:
Thanks for the use case. We should certainly check and filter out multiple components - will apply a fix for this shortly.

However, the case of multiple services using the same source is quite complex as you can only install a single instance of a source. See [How to install multiple windows services with one executable](http://stackoverflow.com/questions/9116870/wix-how-to-install-multiple-windows-services-with-one-executable) & [Using Wix to install process with multiple windows services](http://stackoverflow.com/questions/19414695/using-wix-to-install-process-with-multiple-windows-services).